pypdf/_writer.py

```diff
@@ -369,6 +367,9 @@ def __exit__(
         """Write data to the fileobj."""
         if self.fileobj:
             self.write(self.fileobj)
+            close_attr = getattr(self.fileobj, "close", None)
+            if callable(close_attr):
+                self.fileobj.close()  # type: ignore[attr-defined]
```

Comment:
Why are you not using `close_attr()` directly here? Then you might even be able to omit the ignore comment.

Reply:
Good idea to avoid the mypy issue.  I do, however, think that the intent is more clearly communicated as written, but I will make the change.